tool/tsh/kube.go

```diff
@@ -330,13 +329,13 @@ func getKubeTLSServerName(tc *client.TeleportClient) string {
 		// If proxy is configured without public_addr set the ServerName to the 'kube.teleport.cluster.local' value.
 		// The k8s server name needs to be a valid hostname but when public_addr is missing from proxy settings
 		// the web_listen_addr is used thus webHost will contain local proxy IP address like: 0.0.0.0 or 127.0.0.1
-		return addSubdomainPrefix(constants.APIDomain, alpnproxy.KubeSNIPrefix)
+		return addSubdomainPrefix(constants.APIDomain, constants.KubeTeleportProxyALPNPrefix)
```

Comment:
@smallinsky Sorry to keep on dragging this on but I keep thinking about compatibility implications of the client changes. They basically mean that tsh 9.0 (or Kube agent 9.0) will not be able to connect to a proxy running version that does not recognize the new prefix unless they upgrade to latest 8.x where we will backport this.

Our [compatibility docs](https://goteleport.com/docs/setup/operations/upgrading/#component-compatibility) guarantee 1-version _backwards_ compatibility, not forward compatibility, so this seems to satisfy it but I think we should still try and make an effort to preserve forwards-compatibility where possible to make upgrades easier and less surprising for users.

With that in mind, I think a more graceful "migration" to this new prefix would be as follows:

- In 8.x and 9.0 we only update the server to recognize the new prefix but don't change clients (tsh/agents) so they keep using the old one.
- After we cut 9.0 and master becomes 10.0, we update the clients to use the new prefix. This way we know for sure that any proxy our clients connect to will recognize it.

WDYT?

Reply:
Make sens,  So actually we  drop support for old  prefix on server side in Teleport 11.0